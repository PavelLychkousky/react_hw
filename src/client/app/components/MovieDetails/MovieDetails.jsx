import React from 'react';

import styles from './MovieDetails.less';
import Logo from '../Logo'
import Button from '../Button'

function MovieDetails(props) {
  return (
  <div className="movie-details">
    <Logo />
    <Button title="search" onClick={() => props.history.push('/search')}/>
    <div className="picture">
      <img src={props.movie.coverUri} />
    </div>
    <div className="movie-description">
      <div className="title">{props.movie.title}</div>
      <div className="rating">{props.movie.rating}</div><br/>
      <div className="year">{props.movie.releaseYear}</div>
      <div className="duration">{props.movie.duration} min</div><br/>
      <div className="description">{props.movie.description}</div><br/>
      <div className="director">Director: {props.movie.director}</div><br/>
      <div className="cast">Cast: Actor 1, Actor 2, Actor 3, Actor 4, Actor 5, Actor 6, Actor 7</div>
    </div>
  </div>
  );
}

export default MovieDetails;