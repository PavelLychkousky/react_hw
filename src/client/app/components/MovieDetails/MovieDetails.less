.movie-details {
  background-color: black;
  padding: 15px 40px;
  overflow: auto;

  .btn {
    position: absolute;
    top: 10px;
    right: 10px;
    background-color: white;
    color: #C21F39;
  }

  .picture {
    float: left;
    max-width: 300px;
    min-width: 100px;
  }
  .movie-description {
    float: left;
    color: white;

    .title {
      color: #C21F39;
      font-size: 18px;
      font-weight: bold;

      display: inline-block;
    }

    .rating {
      color: green;
      font-size: 16px;
      border-radius: 50%;
      width: 30px;
      height: 30px;
      text-align: center;
      line-height: 30px;
      border: 2px solid green;
      display: inline-block;
      margin-left: 25px;
      clear: both;
    }

    .year,
    .duration,
    .description,
    .director,
    .cast{
      padding-top: 30px;
      font-size: 16px;
      font-weight: bold;
      display: inline-block;
    }

    .description {
      color: gray;
      font-weight: normal;
    }


    .duration {
      margin-left: 30px;
    }
    .director,
    .cast {
      font-size: 14px;
      color: gray;
    }



  }
}