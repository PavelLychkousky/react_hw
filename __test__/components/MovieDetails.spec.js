import React from 'react';
import MovieDetails from '../../src/client/app/components/MovieDetails';
import renderer from 'react-test-renderer';

it('Should show movie info', () => {
  let props = {
    movie: {
      id: 1, title: 'Movie 1', releaseYear: 2014, genre: 'Genre name', director: 'director 1', rating: '4.1', duration: 105
    }
  };
  const component = renderer.create(
    <MovieDetails {...props}></MovieDetails>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});