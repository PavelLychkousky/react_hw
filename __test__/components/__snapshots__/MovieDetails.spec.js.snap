// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should show movie info 1`] = `
<div
  className="movie-details"
>
  <div
    className="logo"
  >
    netflixroulette
  </div>
  <div
    className="btn"
    onClick={[Function]}
  >
    search
  </div>
  <div
    className="picture"
  >
    <img
      src={undefined}
    />
  </div>
  <div
    className="movie-description"
  >
    <div
      className="title"
    >
      Movie 1
    </div>
    <div
      className="rating"
    >
      4.1
    </div>
    <br />
    <div
      className="year"
    >
      2014
    </div>
    <div
      className="duration"
    >
      105
       min
    </div>
    <br />
    <div
      className="description"
    />
    <br />
    <div
      className="director"
    >
      Director: 
      director 1
    </div>
    <br />
    <div
      className="cast"
    >
      Cast: Actor 1, Actor 2, Actor 3, Actor 4, Actor 5, Actor 6, Actor 7
    </div>
  </div>
</div>
`;
